enhancements/ibm-public-cloud/ibm-public-cloud-support.md

```diff
@@ -0,0 +1,251 @@
+service using this model. This type of setup allows the hosting of many
+cluster control planes on a single Kubernetes cluster. Only the worker nodes are exclusive
+to an individual user cluster and may exist in a different account than the central
+management cluster.
+
+Supporting this type of deployment in OpenShift 4.x requires a different approach to
+installing OpenShift and managing its control plane. Instead of bootstrapping a cluster 
+control plane from scratch on virtual machines, a cluster control plane can be created 
+by applying a set of manifests to an existing cluster in a new namespace. 
+
+Changes are required in different areas of the product in order to make clusters deployed 
+using this method viable. These include changes to the cluster version operator (CVO), web 
+console, second level operators (SLOs) deployed by the CVO, and RHCOS.
+
+## Motivation
+
+Enable OpenShift 4.x deployment on IBM Cloud.
+
+### Goals
+
+- Enable the deployment of OpenShift 4.x on IBM Public Cloud by providing the IBM
+  Public Cloud team the necessary tools to generate manifests needed for a hosted
+  control plane.
+- Ensure that this deployment model remains functional through regular e2e testing
+  on IBM Public Cloud.
+- Make the necessary product changes to make IBM Public Cloud a fully supported 
+  cloud provider.
+- Ensure that clusters running on IBM Public Cloud are conformant OpenShift clusters
+  (pass Kubernetes and OpenShift conformance e2e tests). Fully document any tests that
+  must be skipped due to differences from a traditional deployment.
+
+### Non-Goals
+
+- Make hosted control planes a supported deployment model outside of IBM Public Cloud.
+- Define the automation needed on the IBM Public Cloud side to provision new clusters.
+- Create an alternate installer for OpenShift.
+
+## Proposal
+
+Support for IBM Public Cloud will be added in 3 phases:
+
+1. Beta: Initial rollout. Enables users to create OpenShift clusters
+   using RHEL 7 worker nodes. Nodes are labeled as both worker and master to support
+   SLOs that have masters in their node selectors. IBM provides out of tree cloud support
+   for storage, service load balancers. Ingress, image registry, storage and auth are 
+   configured after initial provisioning (day 2). Calico is used for the SDN.
+2. GA: In-tree support for ingress, image registry, storage, auth, load balancers.
+3. Post-GA: Add support for RHCOS workers
+
+In more detail:
+
+### Beta
+
+Enables an OpenShift cluster to be hosted on top of a Kubernetes/OpenShift cluster. 
+
+Given a release image, a CLI tool generates manifests that instantiate the control plane
+components so that they can be applied to a namespace. IBM provides the automation
+that applies these manifests and instantiates worker nodes that add themselves to 
+the new cluster. Minting of kubelet certificates for these worker nodes is handled
+by IBM automation.
+
+Components that run on the management cluster include:
```

Comment:
for cluster-monitoring: as apiserver monitoring is effectively being disabled when running in "ROKS mode" what is monitoring control plane components on the management cluster?

Reply:
IBM is running their own monitoring solution on their management/tugboat clusters.